### PR TITLE
merge: esp-idf-svc v0.46.0 into stabl-master 

### PR DESCRIPTION
**Pull request by stabl-gjn**
Merge version/tag [v0.46.0](https://github.com/esp-rs/esp-idf-svc/commit/7dab3058d81b517c7e461a18f0edbeb6fa91d2ca) from upstream into `stabl-master`.

# Note

Did a trial compile with https://github.com/Stabl-Energy/Rust-SBC-Client/pull/23 and it looked promising.

### CODE REVIEW

**Comment by stabl-gjn** [comment]
I only had merge conflicts in `src/wifi.rs` and `src/nvs.rs` (https://github.com/Stabl-Energy/esp-idf-svc/commit/12a095b8cbe552a757b08f62c95b86c5ffab1ce1).

File: src/wifi.rs [hunk 1]
```diff
@@ -1615,7 +1615,8 @@ pub struct ApProbeRequestReceivedData {
 pub enum WifiEvent {
     Ready,
 
-    ScanDone,
+    ScanStarted,
```
Comment (schurwanzn-stabl) [inline src/wifi.rs]: ScanStarted seems to have been removed upstream and erroneously reintroduced here https://github.com/Stabl-Energy/esp-idf-svc/compare/423419acf77b7990a99311b9efeda5e404ac0f35...Stabl-Energy:esp-idf-svc:feature/extended-wifi-event#diff-163838ea4fbc63179426321ccf5a036e1c4cbdd36adcdde9a9e30dbf5ae08c4fL1077-L1079

**Review by schurwanzn-stabl — COMMENTED** [review]
The rest looks good to me!

**Comment by schurwanzn-stabl** [comment]
Not sure why my code comment is marked Outdated...

File: src/wifi.rs (hunk 1 above)
Comment (stabl-gjn) [reply src/wifi.rs]: I have removed the variant.

**Review by schurwanzn-stabl — APPROVED** [review]

**Pull request merged**